cpu_comparisons/umap-cuml.ipynb

```diff
@@ -0,0 +1,463 @@
XddCcJSgbtJyFHU1TPkHDUWLgrnEyJjOMazjBcDQIJ94CrmLUgedhz8PMvsKF7R1HtqQYNPAETWIdGRZ/uZ4TPlNJ/GdbqRkZ4Pyba0m/0kXiT/upO6eWbNpCW1RNWnPibvDSeNdiqvf3MmGBj3G5Iab1tqKVO9HnFTGjqA2HUzHjdA8nVNXQcOpCMqYDS9PQNI1a3yR2/Nvv6V6+iTlX/hNjimbicJSQbfew65OP0xHbSdGXp+CZVoGdy5P0JnBNKsX0e3HXBvFOqcR/zVTsIg09a2JnFcVfmEbimTbGzp5P9acXEDx9Mq5pwUPrVjC6rEJmXT+a3wWmhvu8erSAA4Di66ZTcdcS8ruGj/i5l3x1Ns6Tyl/33PIP3cLji67j5au++7d2HyGEEOK9wgRuCYVC4dEk5b8AN4XD4U3v4JxbgXAoFNoeCoX+eFRaeQzJCIM4ZlIPfRGVShN7ahA1s4jSBYHXlTrN9GVRNnhqAqBZgAEqw8HSqXYuSTYFPvdrpi65igmWHMBKW/C61Yd1MBxgHUzEHQ1AchEME6zKALqjUOXH9BloTgNMP+6F51K/988MPv8jSksMhrZ1U1RfyervbmL6tQsZXhvFaULsQIb+3VmmXuBFmQr/wAhtfxyh+I5pVDwW5uSbqnn4k1vxnNmAV9coGxwkMBSlx1uJcfVceubWEsglMdsGyWWjaLrGvvMXoZwOktOn0hhYQN4w2VAWoDE/QO3eNWj1pzDhWxfi79uH1wywN7IWQ3MQuvjLnDqnjy57P07dieODY3nxzz+n/QtPY1a4KT9jOrHcEBO/dD4tX3iczMZu6tZWse77v6LowxMouXo6W3/4ILlskmmzc3T8pJeis8cRX74fsuA6uZrs2n50rwaNRdhDGVQ0h1bsINceR2XzhxOiHTqOCQFyu0bAqUPWxn9lE65dlaSebD/88ShIdwyQj6dYfsW/s/SBm45+hxNCCCGOgXA43AacfYRt2pEev+F+4xv2ywMXHtWGHkMSMIhjwj7wAkaRjZWGhmsbwLbRACttobsLX/Sd5Q4KowIm2BkOraWAgvK5dD6zHsNO4ptfDJ5qrGg7XWvaqV/oI5fMg/+1AYMGVuYtWqIDNrpJoaKSpqG5Rrt9Pgb7X2TEW4kn0cbOV+ax+dsv0Hu2j8nzyvAl9jC+HkodPoxqH13tivTOGCM5Re3iYqIvp1l1/Q4mneQlGbEIXDkJ78IK3HoeXzbN4PoRBv64j8DvzoNknqi/hKFHdhO5dwVjb52JmjUTLBuKqnhk4bdJDY0wdtXl5PQ4RPYx3O1l80+fx3NxI5296xh3+hx0DEyHk8DEOnaEX8Xb3UfHQ7sYWt5F5Z0nY0WzZE2DwPwp9MT2kE0nmHvLp8gucTAxeR71pyzA4fTiPMXDc6d9jeaWNPr4IhJrOyi7/URiv92NEXRhTCum9Jrp6F6T3qteKFzJIifOpmJGfrilcP2cDtyLK9BcOo7JJZRcM42u85+g7xPLyfe+aY2awiUfSdD60HLU/d+URd6EEEKI9wkJGMRRp5QNbc/i8Js4/E4sqxAsZIZ1cr0J/NO8gIZujM6IsxOF26I6MvkElpXCO7iVurkGGn7AhlQXuqnhDmiAhqvSBZpjdKE2wPRAPn54mYDR9RoyvWkcQZNc2iYxkKN0nKcwT//geg6ZAcpqzkdhErhiOmN7FrG7rIqpE3RK83HshEnNSZCKa0wI5WiOllLjSWFWFuO/IUjsy6vp3ZcFa4jMi3HcncMwexFm+wgDsxopqa4lW+XC3p7DjkbJtkTxnt+I84zx5HMKhwbpeAeOGQGcdVV0X/o0NbfXs3/JqQwObGfah032PjzIPscLlG7dh6c6yISPnMSB1B4iGw6Q3h5HzarCeng/6Y19+M4ay/Bd24jcuxPdaVI1YzJDZyVw+TT4QCkRc5jO8Mu48BOYPob6rxbR7S+j99qXMGt9qEye5LI2nAsriD+8j9zOw7lfmtsg8sDuwx90Nkd6eVdhW6kLbXRQJ7Np4G37SDaawFVc9I90LyGEEEK8yySHQRx90bZC1SMMwMYwQDM0PBUQmO7Dtka/rXuqKKyuPiodocfjoj9QDiqPjkKrWgDFEwADTdOomOJD05yF4kjqNSV6vFVguMknINOTQY3GwsOrIwy8MITDZ1La6OHNFPSsQou2YKx6mMlzEvxL3wrGVmRxj6uj/HQfxePcjFnoQUdn4Y3XM/Vr15Pc3UNlfAg94KJ4fhnGkikkDBfOby2mxHSya68L2+tgTpNiklXNOGcJ+Z1Rym9fRPUZdQTveAVXcy8poCubZ9KtjZR/ronyyRPxzq1mIN0Ouyz2L8tz6n03UjJ1DC33/Yn+5csZfvjblHRsY+Cbq0gFDfwfmcC4/72U4Kenkl7fS/KR/bA3zbk/vhXV5CTnyFLvm4oWU+z93hP0XvcS/bevZWj7PqKv+Om+8C/g0kmv6iHXlsCo9lD9q9MJ3jyffGchmKu6ZylmfRGquzBy4D2/AeeM4OGrOJwhsawVAP9HmzBqvX+1i/Su2P53dCghhBBCHE8ywiCOOtX6LOTjoDlQVp5cxMJZerirGQfvpnopRA4mODyQizFmOA66s7DdDIA7CNFWID9aetXmtbU8N9/TS++2JOf8sDD1yHZU4qrpQ0MDdKovqQBMot1JvEED02WAVhhkUEDc4cafS2FZCk90gFSfA9OjAxp7myPUltpsfTRCcJyThgV+mr/ybRKDCkNLMLK8neDNc8jNraDzqQOMmT6Z1Moe2h/ZSdFdZxK9ZydWMEY/W/F3d1PT7iC48FxqL+xhcIyX7JzpPLhO45TJI7j846nrLmLD7tXs+NwDnPnYbfzh81eSGYqy+JcK39Qaxpxewvhzq9n3fIrIo3soD02k7uJF6F4XoQ8sZsV3b6NkTh3WSRNxTSrF0ejnlH+5jq1DzzP44k5K4wH23reXMx76Flt++zCJRYrWp9aSOxCn+menoBs6kZ9sxzmjgq7LnkR
```

Comment:
I agree with @cjnolet here -- trustworthiness (while not a perfect quality measure) is probably a more meaningful score in this context, and is relatively easy to compute given the scikit-learn implementation (it is just potentially quite time consuming for a large dataset like fashion-mnist). It might be reasonable to compute trustworthiness on just the test set (small enough that the sklearn implementation may not choke, since it computes an all-pairs distance matrix).

Reply:
Thank you @cjnolet @lmcinnes .  @cjnolet suggested I do this prior to me post in the notebook and I neglected to do it.  I'll measure those values tonight/tomorrow morning

Reply:
And thank you for the detailed explanation as well.  I was not very familiar with this scoring